PADC_FP_Travelling/app/src/main/java/com/padc/travelling/activities/HomeActivity.java

```diff
@@ -63,6 +79,28 @@ public void onClick(View view) {
     public boolean onCreateOptionsMenu(Menu menu) {
         // Inflate the menu; this adds items to the action bar if it is present.
         getMenuInflater().inflate(R.menu.menu_home, menu);
+
+        MenuItem searchMenuItem = menu.findItem(R.id.action_search);
+        if(searchMenuItem != null){
+
+            MenuItemCompat.setOnActionExpandListener(searchMenuItem, new MenuItemCompat.OnActionExpandListener() {
+                @Override
+                public boolean onMenuItemActionExpand(MenuItem item) {
+                    tvSearchTravel.setVisibility(View.VISIBLE);
+                    flContainer.setVisibility(View.INVISIBLE);
```

Comment:
Are you sure it should be set INVISIBLE and not GONE? The situation to use INVISIBLE is very rare.